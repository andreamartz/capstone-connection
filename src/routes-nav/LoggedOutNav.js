import React from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import DrawerNavigation from './DrawerNavigation';
import WideNavMenu from './WideNavMenu';

const LoggedOutNav = ({ navData, classes, isSmallScreen }) => {
	if (isSmallScreen) {
		return <DrawerNavigation navData={navData} />;
	}
	return (
		<Toolbar className={classes.toolbar}>
			<WideNavMenu navData={navData} classes={classes} />
		</Toolbar>
	);
};

export default LoggedOutNav;
