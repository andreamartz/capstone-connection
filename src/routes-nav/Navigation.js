import React, { useContext, useState } from "react";
import { NavLink } from "react-router-dom";
import UserContext from "../auth/UserContext";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import "./Navigation.css";
import useMediaQuery from '@material-ui/core/useMediaQuery';
import DrawerNavigation from './DrawerNavigation';

const useStyles = makeStyles(theme => ({
  icons: {
    fontSize: '1.4rem',
  },
  iconLogo: {
    color: 'yellow',
    fontSize: '3rem',
  },
  navLink: {
    color: '#FFFFFF',
    textDecoration: 'none',
    '&:hover': {
      textDecoration: "underline",
    }
  },
  // toolbar: {
  //   display: "flex",
  //   justifyContent: "space-between"
  // }
}));

/** Navigation bar for site. Shows up on every page.
 *
 * When user is logged in, shows links to main areas of site. When not,
 * shows link to Login and Signup forms.
 *
 * Rendered by App.
 */


function Navigation({ logout }) {
  // const [anchorEl, setAnchorEl] = useState(null);
  const [value, setValue] = useState(0);
  const classes = useStyles();
  const { currentUser } = useContext(UserContext);
  console.debug("Navigation", "CURRENTUSER: ", currentUser);
  // const handleOpenMenu = e => {
  //   setAnchorEl(e.currentTarget);
  // };

  // const handleMenuClose = () => {
  //   setAnchorEl(null);
  // };
  // const handleClickTab = (e, newValue) => {
  //   setValue(newValue);
  // };

  const navData = currentUser?.username
    ? [{
      label: "Projects",
      href: "/projects"
      },
      {
        label: "My Profile",
        href: `/users/${currentUser.username}/settings`
      }]
    : [{
      label: "Login",
      href: "/login"
    },
    {
      label: "Sign Up",
      href: "/signup"
    }]
  ;
  // const loggedInNavData = [
  //   {
  //     label: "Projects",
  //     href: "/projects"
  //   },
  //   {
  //     label: "My Profile",
  //     href: `/users/${currentUser.username}/settings`
  //   }
  // ];
  
  // const loggedOutNavData = [
  //   {
  //     label: "Login",
  //     href: "/login"
  //   },
  //   {
  //     label: "Sign Up",
  //     href: "/signup"
  //   }
  // ]

  const CapstoneConnectionsLogo = (
    <Typography 
      className={classes.logo}
      variant="h6"
      component="h1"
    >
      Capstone Connections
    </Typography>
  );

  // Breakpoints
  const theme = useTheme();
  const isMatch = useMediaQuery(theme.breakpoints.down('sm'));

  const getWideNavMenuButtons = (navData) => {
    return navData.map(({ label, href }) => {
      return (
        <Button key={label} className={classes.navLink}>
          <NavLink to={href} className={classes.navLink}>
            {label}
          </NavLink>

        </Button>
      )
    });
  }

  const loggedInNav = () => {
    if (isMatch) {
      return (<DrawerNavigation navData={navData} logout={logout}/>);
    }
    return (
      <Toolbar className={classes.toolbar}>
        <div>
          {getWideNavMenuButtons(navData)}
          <NavLink to="/" onClick={logout} className={classes.navLink}>
            <Button
              color="inherit"
            >
              Logout
            </Button>
          </NavLink>

        </div>
      </Toolbar>
    );
  };

  const loggedOutNav = () => {
    if (isMatch) {
      return (
        <DrawerNavigation navData={navData} />
      );
    }
    return (
      <Toolbar className={classes.toolbar}>
        <div>
          {getWideNavMenuButtons(navData)}
        </div>
      </Toolbar>
    )
  }

  return (
    <>
      <AppBar color="primary">
        <Toolbar>
          {CapstoneConnectionsLogo}
          {currentUser ? loggedInNav() : loggedOutNav()}
          {/* {loggedOutNav()} */}
        </Toolbar>
      </AppBar>
    </>
  )
}

export default Navigation;