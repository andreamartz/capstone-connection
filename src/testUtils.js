import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  bio: "This is my bio",
  email: "test@test.net",
  photoUrl: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
