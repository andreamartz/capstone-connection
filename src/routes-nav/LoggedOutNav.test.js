import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import LoggedOutNav from './LoggedOutNav';
import Navigation from './Navigation';
import { UserProvider } from '../testUtils';

describe('User logged in', function () {
	test('it renders without crashing', function () {
		render(
			<MemoryRouter>
				<UserProvider currentUser={null}>
					<Navigation>
						<LoggedOutNav />
					</Navigation>
				</UserProvider>
			</MemoryRouter>,
		);
	});

	test('it matches snapshot', function () {
		const { asFragment } = render(
			<MemoryRouter>
				<UserProvider currentUser={null}>
					<Navigation>
						<LoggedOutNav />
					</Navigation>
				</UserProvider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
