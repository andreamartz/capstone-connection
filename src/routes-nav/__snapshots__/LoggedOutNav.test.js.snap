// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User logged in it matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h1
        class="MuiTypography-root MuiTypography-h6"
      >
        Capstone Connections
      </h1>
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-8 MuiToolbar-gutters"
      >
        <div>
          <button
            aria-label="navigation link"
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-7"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                aria-current="page"
                class="makeStyles-navLink-7 active"
                href="/"
              >
                Home
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="navigation link"
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-7"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                class="makeStyles-navLink-7"
                href="/login"
              >
                Login
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="navigation link"
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-7"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                class="makeStyles-navLink-7"
                href="/signup"
              >
                Sign Up
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
